pkg/networkpolicy/controller.go

```diff
@@ -767,6 +782,74 @@ func (c *Controller) syncNFTablesRules(ctx context.Context) error {
 	return nil
 }
 
+// To avoid a kernel bug caused by UDP DNS request racing with conntrack
+// process the DNS packets only on the PREROUTING hook after DNAT happens
+// so we can see the resolved destination IPs, typically the ones of the Pods
+// that are used for the Kubernetes DNS Service.
+// xref: https://github.com/kubernetes-sigs/kube-network-policies/issues/12
+// This can be removed once all kernels contain the fix in
+// https://github.com/torvalds/linux/commit/8af79d3edb5fd2dce35ea0a71595b6d4f9962350
```

Comment:
(https://github.com/torvalds/linux/commit/8af79d3edb5fd2dce35ea0a71595b6d4f9962350 for reference)